vm/actor/src/builtin/market/mod.rs

```diff
@@ -195,184 +207,273 @@ impl Actor {
         }
 
-        for deal in &params.deals {
-            // Check VerifiedClient allowed cap and deduct PieceSize from cap.
-            // Either the DealSize is within the available DataCap of the VerifiedClient
-            // or this message will fail. We do not allow a deal that is partially verified.
-            if deal.proposal.verified_deal {
-                let ser_params = Serialized::serialize(&BytesParams {
-                    address: deal.proposal.client,
-                    deal_size: BigInt::from(deal.proposal.piece_size.0),
-                })?;
-                rt.send(
-                    *VERIFIED_REGISTRY_ACTOR_ADDR,
-                    VerifregMethod::UseBytes as u64,
-                    ser_params,
-                    TokenAmount::zero(),
-                )?;
-            }
-        }
+        let mut resolved_addrs = HashMap::<Address, Address>::with_capacity(params.deals.len());
+        let baseline_power = request_current_baseline_power(rt)?;
+        let network_qa_power = request_current_network_qa_power(rt)?;
 
         let mut new_deal_ids: Vec<DealID> = Vec::new();
         rt.transaction(|st: &mut State, rt| {
-            let mut prop = Amt::load(&st.proposals, rt.store())
-                .map_err(|e| ActorError::new(ExitCode::ErrIllegalState, e.into()))?;
-            let mut deal_ops = SetMultimap::from_root(rt.store(), &st.deal_ops_by_epoch)
-                .map_err(|e| ActorError::new(ExitCode::ErrIllegalState, e.into()))?;
-
-            for mut deal in params.deals {
-                validate_deal(rt, &deal)?;
+            let mut msm = st.mutator(rt.store());
+            msm.with_pending_proposals(Permission::Write)
+                .with_deal_proposals(Permission::Write)
+                .with_deals_by_epoch(Permission::Write)
+                .with_escrow_table(Permission::Write)
+                .with_locked_table(Permission::Write)
+                .build()
+                .map_err(|e| actor_error!(ErrIllegalState; "failed to load state: {}", e))?;
+
+            for deal in &mut params.deals {
+                validate_deal(rt, &deal, &baseline_power, &network_qa_power)?;
 
                 if deal.proposal.provider != provider && deal.proposal.provider != provider_raw {
-                    return Err(ActorError::new(
-                        ExitCode::ErrIllegalArgument,
-                        "Cannot publish deals from different providers at the same time."
-                            .to_owned(),
-                    ));
+                    return Err(actor_error!(ErrIllegalArgument;
+                        "cannot publish deals from different providers at the same time."));
                 }
 
                 let client = rt.resolve_address(&deal.proposal.client)?.ok_or_else(|| {
                     actor_error!(ErrNotFound;
                         "failed to resolve provider address {}", provider_raw)
```

Comment:
I think the log is wrong here. Should be the client address, not the provider address